### PR TITLE
fix(qwik-city): service worker over-fetching

### LINKED ISSUE #6795: [🐞] 1.8.0 regressions (slow service worker and larger app)
### Which component is affected?

Qwik Runtime

### Describe the bug

I tried to update to qwik from 1.7.2 to 1.8.0 today (no other deps were updated). 

TLDR: In my opinion it seems like the service worker is loading too much unrelated code on some interactions.

- I'm getting up to 2.5 second delays on chrome 127 (6 seconds on firefox 127) for files to load from the service worker when interacting with an "add to cart" button on the page. If the service worker is enabled and i click the button, 132 requests and 103kb is loaded in the network tab. If I disable the service worker completely and click the button, only 44 requests and 67kb is loaded in the network tab and my app interactions go back to being instant.

- File size of my dist/build folder went from `1.09MB / 483 files` to `1.25MB / 662 files`. I haven't found any server code leaking to the browser but its possible I'm missing it.

43 of my 101 e2e tests are no longer passing. Seems related to the much slower service worker because playwright is timing out after 5 seconds while trying to find elements after interacting.

Service worker usage on the chrome storage tab went from ~226kb on first page load to ~600kb. Maybe irrelevant.

I tried deleting my node_modules folder and used `pnpm i --force`. 

I tried setting moduleResolution to Bundler.

I also reverted back to qwik 1.7.2 and my app went back to working as it expected.

Not sure how i can create a reproduction of this problem because its only an issue with some interactions in my ecommerce website, not all.  So I'm hoping someone else might be experiencing this issue or someone can confirm its a 1.8.0 regression.

### Reproduction

none yet

### Steps to reproduce

_No response_

### System Info

```shell
System:
    OS: Windows 10 10.0.19045
    CPU: (20) x64 12th Gen Intel(R) Core(TM) i9-12900HK
    Memory: 10.85 GB / 31.68 GB
  Binaries:
    Node: 20.14.0 - C:\Program Files\nodejs\node.EXE
    npm: 10.7.0 - C:\Program Files\nodejs\npm.CMD
    pnpm: 9.5.0 - C:\Program Files\nodejs\pnpm.CMD
    bun: 1.1.23 - ~\.bun\bin\bun.EXE
  Browsers:
    Edge: Chromium (127.0.2651.74)
    Internet Explorer: 11.0.19041.4355
  npmPackages:
    @builder.io/partytown: ^0.10.2 => 0.10.2
    @builder.io/qwik: 1.8.0 => 1.8.0
    @builder.io/qwik-city: 1.8.0 => 1.8.0
    typescript: 5.4.5 => 5.4.5
    vite: 5.3.5 => 5.3.5
```

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by shairez**
# What is it?

- [ ] Feature / enhancement
- [x] Bug
- [ ] Docs / tests / types / typos
- [ ] Infra

# Description

Fixing over-prefetching by the service worker

Fixes #6795

# Checklist:

- [ ] My code follows the [developer guidelines of this project](https://github.com/QwikDev/qwik/blob/main/CONTRIBUTING.md)
- [ ] I have performed a self-review of my own code
- [ ] I have ran `pnpm change` and documented my changes
- [ ] I have made corresponding changes to the Qwik docs
- [ ] Added new tests to cover the fix / functionality

### CODE REVIEW

**Review by gioboa — DISMISSED** [review]
That's great @shairez 👏

**Comment by shairez** [comment]
Thanks @gioboa !
not ready yet, still have more work to do 😊

**Review by wmertens — APPROVED** [review]
LGTM

File: packages/qwik-city/src/buildtime/runtime-generation/generate-service-worker.ts [hunk 1]
```diff
@@ -38,6 +38,9 @@ export function prependManifestToServiceWorker(
     return null;
   }
 
+  // TODO: add dynamic imports that don't import routes
+  // (anything that doesn't contain _hw export)
+
```
Comment (ianlet) [inline packages/qwik-city/src/buildtime/runtime-generation/generate-service-worker.ts:43]: @shairez do you want to keep that `TODO` here or should we open a new issue for it? Considering caching the dynamic imports is per se an additional improvement to the (current) prefetching.

Comment (ianlet) [reply packages/qwik-city/src/buildtime/runtime-generation/generate-service-worker.ts:43]: oh well I guess it will stay here for now :sweat_smile:

Comment (shairez) [reply packages/qwik-city/src/buildtime/runtime-generation/generate-service-worker.ts:43]: yeah 😅

I'll remove this comment in a different PR

**Pull request merged**